File: 792-BinarySearch/792-BinarySearch.c
// Last Updated: 2/5/2026, 9:22:58 AM
int search(int* nums, int numsSize, int target) {
    int low=0;
    int high=numsSize-1;
    while(low<=high)
    {
        int mid=low+(high-low)/2;
        if(target==nums[mid])
        return mid;
        else if(target>nums[mid])
        low=mid+1;
        else
        high=mid-1;
    }
    return -1;
    
}

File: 792-BinarySearch/792-BinarySearch.py
# Last Updated: 2/5/2026, 9:23:03 AM
class Solution:
    def search(self, arr: List[int], target: int) -> int:
        low,high=0,len(arr)-1
        while low<=high:
            mid=(low+high)//2
            if arr[mid]==target:
                return mid
            if arr[mid]>target:
                high=mid-1
            else:
                low=mid+1
        return -1
        